app/Jobs/UpdateWiki.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Libraries\OsuWiki;
+use App\Libraries\WikiRedirect;
+use App\Models\NewsPost;
+use App\Models\Wiki\Image;
+use App\Models\Wiki\Page;
+use App\Models\Wiki\WikiObject;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+
+class UpdateWiki implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable;
+
+    /** @var string */
+    private $oldHash;
+
+    /** @var string */
+    private $newHash;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param string $oldHash
+     * @param string $newHash
+     * @return void
+     */
+    public function __construct($oldHash, $newHash)
+    {
+        $this->oldHash = $oldHash;
+        $this->newHash = $newHash;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $files = OsuWiki::getUpdatedFiles($this->oldHash, $this->newHash);
+
+        foreach ($files as $file) {
+            $status = $file['status'];
+
+            $object = $this->getObject($file['filename']);
+
+            if ($object) {
+                // because otherwise we'd have newsposts being pointlessly removed and readded into the database
+                if (get_class($object) !== 'NewsPost' || (get_class($object) === 'NewsPost' && $status !== 'removed')) {
+                    $object->forget();
```

Comment:
Is this issue holding the PR back? Because I've looked into this, and:

* for wiki pages, both `forget()` and `get()` dispatch the respective jobs on the queue, and as far as I know they should be executed in the order they were dispatched,
* for images, the functions are completely synchronous, so it shouldn't be an issue at all.

Or do you mean that I should change the wiki page jobs to be handled synchronously in here?

Reply:
looks like what I observed was something slightly different. If the page previously indexed, the index will be deleted by `forget` but as it's done later in queue, when `get` is called, the page is still in index and thus isn't updated.

Reply:
BTW, I wouldn't assume that there will only ever be a single job worker running (and by extension that things will run in order).

If we ever need to scale up the number of workers in the future, jobs will be running concurrently and not necessarily in the order they were created.

Reply:
yeah, I see now, it was because `Page::get()` won't dispatch an index if a page is already in index